src/adapter/src/coord.rs

```diff
@@ -2532,11 +2540,53 @@ impl Coordinator {
             .await
             .unwrap_or_terminate("cannot fail to create collections");
 
+        self.bootstrap_builtin_continual_tasks(new_builtin_continual_tasks)
+            .await;
+
         if !self.controller.read_only() {
             self.apply_local_write(register_ts).await;
         }
     }
 
+    /// TODO(ct1): Hack to make as_of selection happy. It assumes that anything
+    /// with a zero upper has the since set to the initial as_of. Ideally we'd
+    /// write this down in the durable catalog, but that's hard because of boot
+    /// ordering and it's possible that a better long-term fix would be in as_of
+    /// selection itself.
+    async fn bootstrap_builtin_continual_tasks(
+        &mut self,
+        mut collections: Vec<(GlobalId, CollectionDescription<Timestamp>)>,
+    ) {
+        for (id, collection) in &mut collections {
+            let entry = self.catalog.get_entry(&*id);
+            let ct = match &entry.item {
+                CatalogItem::ContinualTask(ct) => ct.clone(),
+                _ => unreachable!("only called with continual task builtins"),
+            };
+            let debug_name = self
+                .catalog()
+                .resolve_full_name(entry.name(), None)
+                .to_string();
+            let (_optimized_plan, physical_plan, _metainfo) = self
+                .optimize_create_continual_task(&ct, *id, self.owned_catalog(), debug_name)
+                .expect("builtin CT should optimize successfully");
+
+            // Determine an as of for the new continual task.
+            let mut id_bundle = dataflow_import_id_bundle(&physical_plan, ct.cluster_id);
+            // Can't acquire a read hold on ourselves because we don't exist yet.
+            id_bundle.storage_ids.remove(&id);
+            let read_holds = self.acquire_read_holds(&id_bundle);
+            let as_of = read_holds.least_valid_read();
```

Comment:
What happens if the CT only depends on itself and then `id_bundle.storage_ids` is empty? It seems like that would end up with `as_of` being `T::minimum()`, which I think is okay?

Reply:
CTs are required to have at least one input. Plus setting the collection.since to `T::minimum()` is what we're doing now, so at least it can't make things any worse